client/state/media/reducer.js

```diff
@@ -2,40 +2,79 @@
  * External dependencies
  */
 import { combineReducers } from 'redux';
-import { keyBy, omit } from 'lodash';
+import { omit } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import { MEDIA_DELETE, MEDIA_RECEIVE } from 'state/action-types';
+import {
+	MEDIA_DELETE,
+	MEDIA_RECEIVE,
+	MEDIA_REQUEST_FAILURE,
+	MEDIA_REQUESTING } from 'state/action-types';
 import { createReducer } from 'state/utils';
+import MediaQueryManager from 'lib/query-manager/media';
 
-export const items = createReducer( {}, {
-	[ MEDIA_RECEIVE ]: ( state, action ) => {
-		const { siteId, media } = action;
+export const queries = ( () => {
+	function applyToManager( state, siteId, method, createDefault, ...args ) {
+		if ( ! state[ siteId ] ) {
+			if ( ! createDefault ) {
+				return state;
+			}
+
+			return {
+				...state,
+				[ siteId ]: ( new MediaQueryManager() )[ method ]( ...args )
+			};
+		}
+
+		const nextManager = state[ siteId ][ method ]( ...args );
+
+		if ( nextManager === state[ siteId ] ) {
+			return state;
+		}
+
+		return {
+			...state,
+			[ siteId ]: nextManager
+		};
+	}
```

Comment:
@aduth would you agree that this might be a good place to look into using the `keyedReducer()`?

https://github.com/Automattic/wp-calypso/blob/024095d874125c0d72a42bdedc4a4856d09695d1/client/state/utils.js#L77

I'm a bit distracted by some of the code, but @iseulde the idea here is that we can identify shape of the state for an item in a collection and then compose that item reducer with this utility to keep things well-separated and keep the shape of the underlying data transparent (vs. hidden behind `initialState = {}`)